table/sst_file_reader.cc

```diff
@@ -95,6 +103,22 @@ Iterator* SstFileReader::NewIterator(const ReadOptions& roptions) {
   return res;
 }
 
+std::unique_ptr<Iterator> SstFileReader::NewTableIterator() {
```

Comment:
Can we have read_options supported for NewTableIterator just like NewIterator. We can perform range reads here as well by exposing read options.

Reply:
That's a good point! It will be great if the iterator created could have configurable behavior. However, I didn't directly add `ReadOptions` as an argument here because the options there are for tuning the behavior of reading from the DB level. Those options don't apply for reading at a raw table level.

For the iterator returned by `NewTableIterator`,  the `Iterator::Seek*()` type of APIs will be expecting internal RocksDB keys, it's a good point I need to think of some public APIs to allow users to scan a specific range. When these APIs are available, you can delegate iterate a specific range to those APIs, with the help of some bound check on your side. Here are some examples of how sst_dump does it, which you maybe you are already familiar with: https://github.com/facebook/rocksdb/blob/d780e7a5610a7b1d48a4787e1c85a2169f241d98/table/sst_file_dumper.cc#L512

Reply:
makes sense !!